### PR TITLE
[13.0] [REF] Purchase Stock: improve performances of PO's computation on lot

### PR DESCRIPTION
**Pull request by ThomasBinsfeld**
Benefit from mass optimization with a read group in the computation of purchase orders of lots.

--------
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by yelizariev** [comment]
@amoyaux

File: addons/purchase_stock/models/stock.py [hunk 1]
```diff
@@ -192,13 +192,23 @@ class ProductionLot(models.Model):
...
+            groupby=['lot_id', 'move_id'],
+            lazy=False,
+        )
+        stock_move_ids_by_lot = {}
+        for dic in read_group_res:
+            lot_id = dic['lot_id'][0]
+            move_id = dic['move_id'][0]
+            stock_move_ids_by_lot.setdefault(lot_id, set())
+            stock_move_ids_by_lot[lot_id].add(move_id)
+        stock_move_model = self.env['stock.move']
```
Comment (amoyaux) [inline addons/purchase_stock/models/stock.py:208]:
```suggestion
        read_group_res = self.env['stock.move.line'].read_group(
            [('lot_id', 'in', self.ids), ('state', '=', 'done'), ('picking_id.location_id.usage', '=', 'supplier')],
            ['lot_id', 'move_id'],
            ['lot_id', 'move_id'],
            lazy=False,
        )
        stock_move_ids_by_lot = {}
        for group in read_group_res:
            lot_id = group['lot_id'][0]
            move_id = group['move_id'][0]
            stock_move_ids_by_lot.setdefault(lot_id, set())
            stock_move_ids_by_lot[lot_id].add(move_id)
```

File: addons/purchase_stock/models/stock.py [hunk 2]
```diff
@@ -192,13 +192,23 @@ class ProductionLot(models.Model):
...
         for lot in self:
-            stock_moves = self.env['stock.move.line'].search([
-                ('lot_id', '=', lot.id),
-                ('state', '=', 'done')
-            ]).mapped('move_id')
-            stock_moves = stock_moves.search([('id', 'in', stock_moves.ids)]).filtered(
-                lambda move: move.picking_id.location_id.usage == 'supplier' and move.state == 'done')
+            lot_stock_move_ids = list(stock_move_ids_by_lot.get(lot.id, set()))
+            stock_moves = stock_move_model.browse(lot_stock_move_ids)
```
Comment (amoyaux) [inline addons/purchase_stock/models/stock.py:211]:
```suggestion
            stock_moves = self.env['stock.move'].browse(lot_stock_move_ids)
```

**Review by amoyaux — COMMENTED** [review]
Okay, I would change the syntax a little bit but the idea is great

**Comment by lathama** [comment]
@ThomasBinsfeld knowing that 13 is no longer updated, is this PR still needed?

**Comment by C3POdoo** [comment]
Dear @ThomasBinsfeld,

Thank you for your contribution but the version 13.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**